src/me/mrCookieSlime/Slimefun/api/Slimefun.java

```diff
@@ -62,7 +62,7 @@ public static boolean hasUnlocked(Player p, ItemStack item, boolean message) {
 		SlimefunItem sfItem = SlimefunItem.getByItem(item);
 		if (sfItem == null) {
 			if (SlimefunItem.isDisabled(item)) {
-				if (message && !(sfItem instanceof VanillaItem)) Messages.local.sendTranslation(p, "messages.disabled-item", true);
+				if (message && SlimefunItem.getState(item).equals(State.DISABLED)) Messages.local.sendTranslation(p, "messages.disabled-item", true);
```

Comment:
You can save a lot of memory by storing the State in a variable and then doing state != ENABLED instead of isDisabled and then doing state != VANILLA for the message thing.